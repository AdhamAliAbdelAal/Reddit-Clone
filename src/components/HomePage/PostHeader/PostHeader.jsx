// mui components
import {
  Avatar, Box, Typography,
} from '@mui/material';
import JoinButton from '../../JoinButton/JoinButton';

// styles
import {
  Flair,
  PostInfo, PostInfoLink, PostTitle,
} from './styles';

function PostHeader(props) {
  const {
    title, image, sr, creator, flair, flairBackgroundColor, flairColor,
  } = props;
  console.log(flairColor);
  return (
    <>
      <PostInfo py={1}>
        <Avatar
          src={image}
          sx={{
            width: 20,
            height: 20,
          }}
          alt="Profile Image"
        />
        <PostInfoLink to="/" color="#000" fontWeight="bolder">
          r/
          {sr}
        </PostInfoLink>
        <Box color="#787C7E" fontWeight={300} display="flex" gap="4px" flexWrap="wrap">
          <span>
            â€¢
          </span>
          <div>Posted By</div>
          <PostInfoLink to="/" color="inherit" fontWeight="normal">
            u/
            {creator}
          </PostInfoLink>
        </Box>
        <Box display="flex" justifyContent="flex-end" flexGrow={1} alignItems="flex-start">
          <JoinButton />
        </Box>
      </PostInfo>
      <PostTitle to="/">
        <Typography
          variant="h6"
          component="h3"
          sx={{ fontSize: 18 }}
        >
          {title}
          {' '}
          <Flair
            disableRipple
            backgroundColor={flairBackgroundColor}
            flairColor={flairColor}
          >
            {flair}
          </Flair>
        </Typography>
      </PostTitle>
    </>
  );
}

export default PostHeader;
