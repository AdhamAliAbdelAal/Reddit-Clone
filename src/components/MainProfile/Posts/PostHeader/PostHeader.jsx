import { Box, Typography } from '@mui/material';
import moment from 'moment/moment';
import LockIcon from '@mui/icons-material/Lock';
import Inventory2Icon from '@mui/icons-material/Inventory2';
import DoDisturbAltIcon from '@mui/icons-material/DoDisturbAlt';
import CheckCircleIcon from '@mui/icons-material/CheckCircle';
import CallSplitIcon from '@mui/icons-material/CallSplit';

import { useState } from 'react';
import PostJoin from '../../../SubReddit/PostJoin';
import {
  HeaderAvatar, HeaderAvatarImage, HeaderPost, LinkTo, Joined, LinkCross,
} from './styles';

/**
 * Header for a post
 *
 * @component PostHeader
 * @property {string} subReddit - name of subReddit the post published in
 * @property {string} nameuser - name of publisher
 * @property {string} time - time of publishing the post
 * @returns {React.Component} PostHeader
 */

function PostHeader(props) {
  const {
    subReddit, nameUser, Time, type, icon, locked, modState, notJoined, sharedFrom,
  } = props;

  const [joined, setJoined] = useState(false);

  const handleJoin = () => {
    PostJoin(`/subreddits/${subReddit}/subscribe`, joined ? 'unsub' : 'sub');
    setJoined((prev) => !prev);
  };

  const [hover, setHover] = useState(false);

  const handleMouseIn = () => {
    setHover(true);
  };
  const handleMouseOut = () => {
    setHover(false);
  };

  return (
    <HeaderPost>
      <Box sx={{
        display: 'flex',
        flexWrap: 'wrap',
      }}
      >
        <HeaderAvatar>
          <HeaderAvatarImage src={icon} />
        </HeaderAvatar>

        <LinkTo to={(type === 'Subreddit') ? `/r/${subReddit}` : `/user/${subReddit}`}>
          <Typography variant="caption" sx={{ fontWeight: 700, '&:hover': { textDecoration: 'underline' } }}>
            {type === 'Subreddit' ? 'r/' : 'u/'}
            {subReddit}
            {' '}
            .
            {' '}
          </Typography>
        </LinkTo>
        <Typography variant="caption" sx={{ color: '#787c7e', marginLeft: 1 }}>
          {' '}
          Posted by
        </Typography>
        <LinkTo to={`/user/${nameUser}`}>
          <Typography variant="caption" sx={{ color: '#787c7e', '&:hover': { textDecoration: 'underline' } }}>
            u/
            {nameUser}
          </Typography>
        </LinkTo>
        {sharedFrom && (
        <>
          <CallSplitIcon sx={{ color: '#0079D3', marginLeft: '3px', transform: 'rotate(90deg)' }} fontSize="string" />
          <Typography variant="caption" sx={{ color: '#787c7e', marginLeft: 1 }}>
            cross posted by
            {' '}
            <LinkCross to={`/user/${sharedFrom.author.name}`}>
              {sharedFrom.author.name}
            </LinkCross>
          </Typography>
        </>
        )}
        <Box sx={{ flex: '1 1 auto', display: 'flex' }}>
          <Typography
            variant="caption"
            sx={{
              color: '#787c7e', marginLeft: 1,
            }}
          >
            {' '}
            {(moment.utc(Time).local().startOf('seconds')
              .fromNow())}
          </Typography>

          {locked && <LockIcon sx={{ color: '#ffd635', marginLeft: '3px' }} fontSize="string" />}
          {modState === 'spammed' && <Inventory2Icon sx={{ color: '#ff585b', marginLeft: '3px' }} fontSize="string" />}
          {modState === 'approved' && <CheckCircleIcon sx={{ color: '#75d377', marginLeft: '3px' }} fontSize="string" />}
          {modState === 'removed' && <DoDisturbAltIcon sx={{ color: '#ff585b', marginLeft: '3px' }} fontSize="string" />}
        </Box>
        {(type === 'Subreddit' && notJoined) && (
        <Joined
          variant={(joined ? 'outlined' : 'contained')}
          onClick={handleJoin}
          onMouseEnter={handleMouseIn}
          onMouseLeave={handleMouseOut}
          data-testid="join button"
        >
          {(joined ? (hover ? 'leave' : 'joined') : 'join')}
        </Joined>
        )}
      </Box>
    </HeaderPost>
  );
}

export default PostHeader;
