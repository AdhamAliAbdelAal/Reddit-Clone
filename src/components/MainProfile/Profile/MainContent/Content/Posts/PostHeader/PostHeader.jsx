import { Typography } from '@mui/material';
import moment from 'moment/moment';
// import BlockOutlinedIcon from '@mui/icons-material/BlockOutlined';
// import CheckCircleIcon from '@mui/icons-material/CheckCircle';
import {
  HeaderAvatar, HeaderAvatarText, HeaderPost, LinkTo,
} from './styles';

/**
 * Header for a post
 *
 * @component PostHeader
 * @property {string} subReddit - name of subReddit the post published in
 * @property {string} nameuser - name of publisher
 * @property {string} time - time of publishing the post
 * @returns {React.Component} PostHeader
 */

function PostHeader(props) {
  const {
    subReddit, nameUser, Time,
  } = props;
  return (
    <HeaderPost>
      <HeaderAvatar>
        <HeaderAvatarText>r/</HeaderAvatarText>
      </HeaderAvatar>
      <LinkTo to={`/Subreddit/${subReddit}`}>
        <Typography variant="caption" sx={{ fontWeight: 700, '&:hover': { textDecoration: 'underline' } }}>
          r/
          {subReddit}
          {' '}
          .
        </Typography>
      </LinkTo>
      <Typography variant="caption" sx={{ color: '#787c7e', marginLeft: 1 }}>
        Posted by
      </Typography>
      <LinkTo to={`/user/${nameUser}`}>
        <Typography variant="caption" sx={{ color: '#787c7e', marginLeft: 1, '&:hover': { textDecoration: 'underline' } }}>
          u/
          {nameUser}
        </Typography>
      </LinkTo>
      <Typography variant="caption" sx={{ color: '#787c7e', marginLeft: 1 }}>
        {' '}
        {(moment.utc(Time).local().startOf('seconds')
          .fromNow())}
      </Typography>
      {/* {((subTitle === 'Spam').toString() === 'true')
      && (
        <RemovalBox>
          <BlockOutlinedIcon fontSize="string" />
          <Typography variant="caption">Add a removal reason</Typography>
        </RemovalBox>
      )}
      {((subTitle === 'Edited').toString() === 'true')
      && (
      <ApprovedBox>
        <CheckCircleIcon fontSize="string" />
      </ApprovedBox>
      )} */}
    </HeaderPost>
  );
}

export default PostHeader;
