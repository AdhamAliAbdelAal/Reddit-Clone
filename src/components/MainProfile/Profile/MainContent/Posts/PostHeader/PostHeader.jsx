import { Typography } from '@mui/material';

import LockIcon from '@mui/icons-material/Lock';
import Inventory2Icon from '@mui/icons-material/Inventory2';
import moment from 'moment/moment';
import {
  Flair,
  HeaderPost, LinkTo,
} from './styles';

/**
 * Header of the post
 *
 * @component PostHeader
 * @property {string} nameUser -name of the currently loggedin user
 * @property {string} Time -contains the time of the post
 * @property {string} nameUser -contains the name of subreddit the post piblished in
 * @returns {React.Component} PostHeader
 */

function PostHeader(props) {
  const {
    nameUser,
    Time,
    subReddit,
    type,
    nsfw,
    locked,
    flairBackgroundColor,
    flairColor,
    flair,
  } = props;
  return (
    <HeaderPost>
      <LinkTo to={`/Subreddit/${subReddit}`}>
        {type === 'Subreddit' ? (
          <Typography variant="caption" sx={{ fontWeight: 700, '&:hover': { textDecoration: 'underline' } }}>
            r/
            {subReddit}
            {' '}
            .
          </Typography>
        ) : (
          <Typography variant="caption" sx={{ fontWeight: 700, '&:hover': { textDecoration: 'underline' } }}>
            u/
            {subReddit}
            {' '}
            .
          </Typography>
        )}
      </LinkTo>
      <Typography variant="caption" sx={{ color: '#787c7e', marginLeft: 1 }}>
        Posted by
      </Typography>
      <LinkTo to={`/user/${nameUser}`}>
        <Typography variant="caption" sx={{ color: '#787c7e', marginLeft: 1, '&:hover': { textDecoration: 'underline' } }}>
          u/
          {nameUser}
        </Typography>
      </LinkTo>
      <Typography variant="caption" sx={{ color: '#787c7e', marginLeft: 1 }}>
        {' '}
        {(moment.utc(Time).local().startOf('seconds')
          .fromNow())}
      </Typography>
      <Flair
        disableRipple
        backgroundColor={flairBackgroundColor}
        flairColor={flairColor}
      >
        {flair}
      </Flair>
      {locked && <LockIcon sx={{ color: '#ffd635', marginLeft: '3px' }} fontSize="string" />}
      {nsfw && <Inventory2Icon sx={{ color: '#ff585b', marginLeft: '3px' }} fontSize="string" />}

    </HeaderPost>
  );
}

export default PostHeader;
