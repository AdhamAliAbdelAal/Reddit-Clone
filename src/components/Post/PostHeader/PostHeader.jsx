// mui components
import {
  Avatar, Box, Typography,
} from '@mui/material';
import calculateTime from '../../../utils/calculateTime';
import JoinButton from '../../JoinButton/JoinButton';

// styles
import {
  CreatedAt,
  Flair,
  PostInfo, PostInfoLink, PostTitle,
} from './styles';
/**
 * This component is the upper section of post
 *
 * @component PostHeader
 * @property {string} title -Post title.
 * @property {string} image -Post owner icon.
 * @property {string} owner -Post subreddit(post owner).
 * @property {string} author -Post author.
 * @property {string} flairText -Post flair text.
 * @property {string} flairBackgroundColor -Post flair background color.
 * @property {string} flairColor -Post flair color.
 * @property {boolean} subredit -to identify if post in home page or subreddit.
 * @returns {React.Component} Post header
 */

function PostHeader(props) {
  const {
    title, image, owner, author, flair, flairBackgroundColor, flairColor, createdAt,
    subredit,
  } = props;
  return (
    <>
      <PostInfo pb={1}>
        {!subredit
        && (
          <>
            <Avatar
              src={image}
              sx={{
                width: 20,
                height: 20,
              }}
              alt="Profile Image"
            />

            <PostInfoLink to={`/Subreddit/${owner}`} color="#000" fontWeight="bolder">
              r/
              {owner}
            </PostInfoLink>
          </>
        )}
        <Box color="#787C7E" fontWeight={300} display="flex" gap="4px" flexWrap="wrap">
          {!subredit && (
          <span>
            â€¢
          </span>
          )}
          <div>Posted By</div>
          <PostInfoLink to={`/user/${author}`} color="inherit" fontWeight="normal">
            u/
            {author}
          </PostInfoLink>

          <CreatedAt color="inherit" fontWeight="normal">
            {calculateTime(createdAt)}
          </CreatedAt>

        </Box>
        {!subredit && (
        <Box display="flex" justifyContent="flex-end" flexGrow={1} alignItems="flex-start">
          <JoinButton />
        </Box>
        )}
      </PostInfo>
      <PostTitle to="/">
        <Typography
          variant="h6"
          component="h3"
          sx={{ fontSize: 18 }}
        >
          {title}
          {' '}
          <Flair
            disableRipple
            backgroundColor={flairBackgroundColor}
            flairColor={flairColor}
          >
            {flair}
          </Flair>
        </Typography>
      </PostTitle>
    </>
  );
}

export default PostHeader;
