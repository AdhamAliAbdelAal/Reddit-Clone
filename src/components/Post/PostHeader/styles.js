import {
  Box,
} from '@mui/material';
import { styled } from '@mui/system';
import { Link } from 'react-router-dom';
import RedditButton from '../../RedditButton/RedditButton';

export const PostInfo = styled(Box)(() => ({
  display: 'flex',
  fontSize: '12px',
  gap: '4px',
}));

export const PostInfoLink = styled(Link)(({ color, fontWeight }) => ({
  fontWeight,
  color,
  textDecoration: 'none',
  '&:hover': {
    textDecoration: 'underline',
  },
}));

export const PostTitle = styled(Link)(() => ({
  textDecoration: 'none',
  color: '#000',
  '& .MuiTypography-root': {
    lineHeight: 1.2,
    fontWeight: 500,
  },
}));

export const Flair = styled(RedditButton)(({ backgroundColor, flairColor }) => ({
  fontSize: 10,
  padding: '2px 10px',
  fontWeight: 'normal',
  backgroundColor,
  color: flairColor,
  '&:hover': {
    backgroundColor,
  },
}));
