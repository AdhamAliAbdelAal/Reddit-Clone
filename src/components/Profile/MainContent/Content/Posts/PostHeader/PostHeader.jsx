import { Typography } from '@mui/material';
// import BlockOutlinedIcon from '@mui/icons-material/BlockOutlined';
// import CheckCircleIcon from '@mui/icons-material/CheckCircle';
import {
  HeaderAvatar, HeaderAvatarText, HeaderPost,
} from './styles';

/**
 * Header for a post
 * @return {React.Component} - PostHeader
 * @param {string} subReddit - name of subReddit the post published in
 * @param {string} nameuser - name of publisher
 * @param {string} time - time of publishing the post
 */

// calculate the time difference between post creation and current date
const calculateTime = (d, time) => {
  const year = d.getFullYear() - time.split('T')[0].split('-')[0];
  const month = d.getMonth() - time.split('T')[0].split('-')[1];
  const day = d.getDate() - time.split('T')[0].split('-')[2];

  if (year > 0) {
    return (`${year} years ago`);
  }
  if (month > 0) {
    return (`${month} months ago`);
  }
  if (day > 0) {
    return (`${day} days ago`);
  }
  return ('today');
};

function PostHeader(props) {
  const {
    subReddit, nameUser, Time,
  } = props;
  const d = new Date();
  return (
    <HeaderPost>
      {/* OR PHOTO */}
      <HeaderAvatar>
        <HeaderAvatarText>r/</HeaderAvatarText>
      </HeaderAvatar>
      {/* OR PHOTO */}
      <Typography variant="caption" sx={{ fontWeight: 700, marginLeft: 1, '&:hover': { textDecoration: 'underline' } }}>
        r/
        {subReddit}
        {' '}
        .
      </Typography>
      <Typography variant="caption" sx={{ color: '#787c7e', marginLeft: 1 }}>
        Posted by
      </Typography>
      <Typography variant="caption" sx={{ color: '#787c7e', marginLeft: 1, '&:hover': { textDecoration: 'underline' } }}>
        u/
        {nameUser}
      </Typography>
      <Typography variant="caption" sx={{ color: '#787c7e', marginLeft: 1 }}>
        {' '}
        {calculateTime(d, Time)}
      </Typography>
      {/* {((subTitle === 'Spam').toString() === 'true')
      && (
        <RemovalBox>
          <BlockOutlinedIcon fontSize="string" />
          <Typography variant="caption">Add a removal reason</Typography>
        </RemovalBox>
      )}
      {((subTitle === 'Edited').toString() === 'true')
      && (
      <ApprovedBox>
        <CheckCircleIcon fontSize="string" />
      </ApprovedBox>
      )} */}
    </HeaderPost>
  );
}

export default PostHeader;
