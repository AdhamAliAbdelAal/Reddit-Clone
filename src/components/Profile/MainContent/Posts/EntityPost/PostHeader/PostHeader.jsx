import { Typography } from '@mui/material';
// import BlockOutlinedIcon from '@mui/icons-material/BlockOutlined';
// import CheckCircleIcon from '@mui/icons-material/CheckCircle';
import {
  HeaderPost,
} from './styles';

function PostHeader(props) {
  const { nameUser, Time, subReddit } = props;
  return (
    <HeaderPost>
      <Typography variant="caption" sx={{ fontWeight: 700, '&:hover': { textDecoration: 'underline' } }}>
        r/
        {subReddit}
        {' '}
        .
      </Typography>
      <Typography variant="caption" sx={{ color: '#787c7e', marginLeft: 1 }}>
        Posted by
      </Typography>
      <Typography variant="caption" sx={{ color: '#787c7e', marginLeft: 1, '&:hover': { textDecoration: 'underline' } }}>
        u/
        {nameUser}
      </Typography>
      <Typography variant="caption" sx={{ color: '#787c7e', marginLeft: 1 }}>
        {' '}
        {Time}
      </Typography>
      {/* {((subTitle === 'Spam').toString() === 'true')
      && (
        <RemovalBox>
          <BlockOutlinedIcon fontSize="string" />
          <Typography variant="caption">Add a removal reason</Typography>
        </RemovalBox>
      )}
      {((subTitle === 'Edited').toString() === 'true')
      && (
      <ApprovedBox>
        <CheckCircleIcon fontSize="string" />
      </ApprovedBox>
      )} */}
    </HeaderPost>
  );
}

export default PostHeader;
