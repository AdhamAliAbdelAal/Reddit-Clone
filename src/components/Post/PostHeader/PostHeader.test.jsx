import renderer from 'react-test-renderer';
import '@testing-library/jest-dom';
import { MemoryRouter as Router } from 'react-router-dom';
import PostHeader from './PostHeader';
import calculateTime from '../../../utils/calculateTime';

describe(PostHeader, () => {
  it('PostHeader renders correctly', () => {
    const testCase = {
      title: 'title', image: 'image', owner: 'owner', creator: 'creator', flair: 'flair', flairBackgroundColor: 'flairBackgroundColor', flairColor: 'flairColor',
    };
    const tree = renderer
      .create(
        <Router>
          <PostHeader
            title={testCase.title}
            image={testCase.image}
            owner={testCase.owner}
            creator={testCase.creator}
            flair={testCase.flairText}
            flairBackgroundColor={testCase.flairBackgroundColor}
            flairColor={testCase.flairColor}
          />
        </Router>
        ,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe(calculateTime, () => {
  const second = 1000;
  const minute = second * 60;
  const hour = minute * 60;
  const day = hour * 24;
  const week = day * 7;
  const month = day * 30;
  const year = day * 365;

  it('calculateTime works correctly', () => {
    const today = new Date();
    const testCases = [{
      input: today - second,
      output: '1 second',
    },
    {
      input: today - 2 * second,
      output: '2 seconds',
    },
    {
      input: today - 60 * second,
      output: '1 minute',
    },
    {
      input: today - minute,
      output: '1 minute',
    },
    {
      input: today - 2 * minute,
      output: '2 minutes',
    },
    {
      input: today - 120 * minute,
      output: '2 hours',
    },
    {
      input: today - hour,
      output: '1 hour',
    },
    {
      input: today - 2 * hour,
      output: '2 hours',
    },
    {
      input: today - 24 * hour,
      output: '1 day',
    },
    {
      input: today - 1 * day,
      output: '1 day',
    },
    {
      input: today - 2 * day,
      output: '2 days',
    },
    {
      input: today - 7 * day,
      output: '1 week',
    },
    {
      input: today - 14 * day,
      output: '2 weeks',
    },
    {
      input: today - week,
      output: '1 week',
    },
    {
      input: today - 2 * week,
      output: '2 weeks',
    },
    {
      input: today - 30 * day,
      output: '1 month',
    },
    {
      input: today - month,
      output: '1 month',
    },
    {
      input: today - 2 * month,
      output: '2 months',
    },
    {
      input: today - 365 * day,
      output: '1 year',
    },
    {
      input: today - year,
      output: '1 year',
    },
    {
      input: today - 2 * year,
      output: '2 years',
    },
    ];
    testCases.forEach((testCase) => {
      console.log(testCase.input);
      expect(calculateTime(new Date(testCase.input).toISOString())).toBe(testCase.output);
    });
  });
});
