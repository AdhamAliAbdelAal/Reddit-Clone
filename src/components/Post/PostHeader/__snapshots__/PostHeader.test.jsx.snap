// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostHeader PostHeader renders correctly 1`] = `
[
  <div
    className="MuiBox-root css-11kldd9"
  >
    <div
      className="MuiAvatar-root MuiAvatar-circular css-2is7ki-MuiAvatar-root"
    >
      <img
        alt="Profile Image"
        className="MuiAvatar-img css-1pqm26d-MuiAvatar-img"
        src="image"
      />
    </div>
    <a
      className="css-kqg5s"
      color="#000"
      fontWeight="bolder"
      href="/"
      onClick={[Function]}
    >
      r/
      owner
    </a>
    <div
      className="MuiBox-root css-1clzuz0"
    >
      <span>
        â€¢
      </span>
      <div>
        Posted By
      </div>
      <a
        className="css-1fa4cpj"
        color="inherit"
        fontWeight="normal"
        href="/"
        onClick={[Function]}
      >
        u/
        creator
      </a>
    </div>
    <div
      className="MuiBox-root css-m647gw"
    >
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-1b88tlx-MuiButtonBase-root-MuiButton-root"
        data-testid="join button"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        join
      </button>
    </div>
  </div>,
  <a
    className="css-10m9jmh"
    href="/"
    onClick={[Function]}
  >
    <h3
      className="MuiTypography-root MuiTypography-h6 css-6aghek-MuiTypography-root"
    >
      title
       
      <button
        backgroundColor="flairBackgroundColor"
        className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-748rc4-MuiButtonBase-root-MuiButton-root"
        disabled={false}
        flairColor="flairColor"
        onBlur={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      />
    </h3>
  </a>,
]
`;
